api/core/external_data_tool/external_data_fetch.py

```diff
@@ -31,7 +31,6 @@ def fetch(
         :return: the filled inputs
         """
         results: dict[str, Any] = {}
-        inputs = dict(inputs)
```

Comment:
I'm sorry I didn't explain my intention clearly. In this method, we indicate to the caller that the inputs won't be directly modified by marking the parameters as a Mapping type. Then, on line 34, we make a copy of the inputs, and finally, we provide the modified inputs as a return value for the user.